src/contracts.rs

```diff
@@ -583,3 +583,350 @@ impl WatchOnlySwapCoin {
         })
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bitcoin::consensus::encode::deserialize;
+    use bitcoin::hashes::hex::{FromHex, ToHex};
+    use bitcoin::PrivateKey;
+    use rand::{thread_rng, Rng};
+    use std::str::FromStr;
+    use std::string::String;
+
+    // Added as a helper, maybe useful later.
```

Comment:
Nit: this comment is wrong now and should be removed. The function isn't useful later but useful now because it's needed for the tests.

Reply:
Updated..